File: gcd.py
#!/usr/bin/env python3

import sys
import functools


def gcd2(a, b):
    while b != 0:
        a, b = b, a % b
    return a


def gcdn(nums):
    return functools.reduce(gcd2, nums)


if len(sys.argv) > 1:
    print(gcdn(map(int, sys.argv[1:])))

File: gcd.rb
#!/usr/bin/env ruby

def gcd2 a, b
    if b == 0
        a
    else
        gcd2 b, a % b
    end
end

def gcdn ns
    ns.reduce{ |a, b| gcd2 a, b }
end

puts gcdn ARGV.collect{|s| s.to_i}

